### DISCUSSION TITLE
Extracting Current Item in Permutation

### DISCUSSION THREAD

**Discussion by msarvestani** [bonsai-rx/bonsai#1303] [Q&A] [answered]
Hi,
I am creating stimuli in a randomized trial structure where each trial has x conditions that I permute randomly. I use RepeatCount after drawing the stimulus to get y trials. 

My problem is that I also need to keep track of the current item in the permuted set that is being used to update the location of the drawn stimulus on each condition. I need to use this information to close the loop of my experiment so I can't just write it to a file. What is the best way to do this?  If I simply extract the current item from 'Permutation', my workflow stops after 1 trial as if I didn't have a RepeatCount in place.

Thanks so much. 

![image](https://user-images.githubusercontent.com/12864873/227384315-2202c375-573c-493e-89c8-5ffcaaa66256.png)

[debug_stimuli.zip](https://github.com/bonsai-rx/bonsai/files/11057133/debug_stimuli.zip)

**Comment by bruno-f-cruz** [comment, marked as answer]
Not sure I completely understood the scope of the question.
If you want to keep track of the stimuli parameters inside `createobservable` you just need to save your input to a subject. Probably an AsyncSubject to make sure it doesn't get overridden:

![image](https://user-images.githubusercontent.com/7049351/227468146-11985699-c8be-40ea-b4f7-880c00e9f022.png)

Then you can subscribe to this subject anywhere inside the CreateObservable node.

**Reply by msarvestani** [reply]
Ah that's so easy! I was trying to save the input to a subject using Source1's 'Create Source' menu (that comes up with a Right-click on Source1) and it wouldn't let me create a connection. But if I drop AsyncSubject from the Toolbox I can connect it to Source. I thought selecting from the Toolbox and the Source menu were equivalent . Thanks a lot.

**Reply by msarvestani** [reply]
Sorry one more question. How can I make the CurrentPair value available outside of the createobservable?
